common/src/block_interval.rs

```diff
@@ -0,0 +1,255 @@
+use alloy::primitives::B256;
+use alloy::rpc::types::eth::BlockId;
+use alloy::{hex, providers::Provider, transports::Transport};
+use async_stream::try_stream;
+use thiserror::Error;
+use tokio_stream::Stream;
```

Comment:
```suggestion
use futures::Stream;
```

This is the authoritative definition, you probably don't need tokio_stream unless you're dealing with tokio types

Reply:
`tokio_stream` is just re-exporting the `futures_core::Stream` which is the same thing that `futures` library does. One could argue that as I am using `tokio_stream` to generate stream, it is more natural to use `tokio` Stream type. But I will agree with you on this.